helpers/testsuite/testing_helpers.go

```diff
@@ -359,3 +366,157 @@ func cleanCLIOutput(CLIOutput []byte, item string) (cleanedOutput []byte, err er
 
 	return []byte(outputString), nil
 }
+
+// NewConfig returns a config object from the data passed
+func NewConfig(t *testing.T, configBytes []byte) *config.Config {
+	conf, err := config.LoadConfig([]byte(configBytes))
+	if err != nil {
+		t.Fatal("config loading error:", err)
+	}
+	if !conf.Valid() {
+		t.Fatal("config is not valid")
+	}
+	return conf
+}
+
+// The following sign handler functions are only here and not in api/sign
+// to avoid a recursive import of signer/universal.
+
+// NewSignHandlerFromSigner generates a new SignHandler directly from
+// an existing signer.
+func NewSignHandlerFromSigner(s signer.Signer) (h http.Handler, err error) {
+	policy := s.Policy()
+	if policy == nil {
+		err = cferr.New(cferr.PolicyError, cferr.InvalidPolicy)
+		return
+	}
+
+	// Sign will only respond for profiles that have no auth provider.
+	// So if all of the profiles require authentication, we return an error.
+	haveUnauth := (policy.Default.Provider == nil)
+	for _, profile := range policy.Profiles {
+		if !haveUnauth {
+			break
+		}
+		haveUnauth = (profile.Provider == nil)
+	}
+
+	if !haveUnauth {
+		err = cferr.New(cferr.PolicyError, cferr.InvalidPolicy)
+		return
+	}
+
+	return &api.HTTPHandler{
+		Handler: &SignHandler{
+			signer: s,
+		},
+		Methods: []string{"POST"},
+	}, nil
+}
+
+// A SignHandler accepts requests with a hostname and certficate
+// parameter (which should be PEM-encoded) and returns a new signed
+// certificate. It includes upstream servers indexed by their
+// profile name.
+type SignHandler struct {
+	signer signer.Signer
+}
+
+// Handle responds to requests for the CA to sign the certificate request
+// present in the "certificate_request" parameter for the host named
+// in the "hostname" parameter. The certificate should be PEM-encoded. If
+// provided, subject information from the "subject" parameter will be used
+// in place of the subject information from the CSR.
+func (h *SignHandler) Handle(w http.ResponseWriter, r *http.Request) error {
+	log.Info("signature request received")
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return err
+	}
+	r.Body.Close()
+
+	var req signer.SignRequest
+	err = json.Unmarshal(body, &req)
+	if err != nil {
```

Comment:
The more concise version of `if err = json.Unmarshal(body, &req); err != nil {` could be used here as well